src/org/labkey/test/tests/core/admin/AllowedFileExtensionAdminTest.java

```diff
@@ -0,0 +1,618 @@
+        {
+            if (!excludedTypes.contains(entry.getKey()))
+            {
+                _fileBrowserHelper.uploadFile(entry.getValue());
+                expectedFiles.add(entry.getValue().getName());
+            }
+        }
+
+        List<String> actualFiles = _fileBrowserHelper.getFileList();
+
+        checker().withScreenshot()
+                .verifyEqualsSorted("Uploaded files not as expected.",
+                        expectedFiles, actualFiles);
+
+    }
+
+    private void uploadToFileWebPartExcluded(List<String> excludedTypes)
+    {
+
+        goToProjectHome();
+
+        for (String excludedType : excludedTypes)
+        {
+
+            if (_fileBrowserHelper.fileIsPresent(fileMap.get(excludedType).getName()))
+            {
+                log(String.format("Remove the '%s' file from the upload.", excludedType));
+                _fileBrowserHelper.deleteFile(fileMap.get(excludedType).getName());
+                refresh();
+            }
+
+            Window<?> errorWindow = _fileBrowserHelper.uploadFileExpectingError(fileMap.get(excludedType));
+
+            checker().withScreenshot()
+                    .verifyEquals(String.format("Error message for excluded file type '%s' not as expected.", excludedType),
+                            String.format("This file type [%s] is not allowed.", excludedType.replace(".", "")), errorWindow.getBody());
+
+            click(Ext4Helper.Locators.ext4Button("OK"));
+        }
+
+    }
+
+    /**
+     * <p>
+     *     Using a list validate that attachments work correctly with the allowed files list.
+     * </p>
+     * <p>
+     *     After creating a list that has an auto-index and a attachment field as the only field this test will:
+     *     <ul>
+     *         <li>Can insert an element into the list with the allowed file type.</li>
+     *         <li>Validate that a file of the type not allowed is not allowed.</li>
+     *         <li>Remove the bad file and resubmit with a valid file type.</li>
+     *     </ul>
+     * </p>
+     *
+     */
+    @Test
+    public void testAllowedFileExtensionsInLists()
+    {
+
+        List<String> allowedTypes = new ArrayList<>();
+        String excludedType = ".csv";
+
+        for (String extension : fileMap.keySet())
+        {
+            if (!extension.equals(excludedType))
+                allowedTypes.add(extension);
+        }
+
+        log(String.format("Add the following as allowed extensions: %s", allowedTypes));
+        setAllowedExtensions(allowedTypes, allowedTypes);
+
+        goToProjectHome();
+        String listName = "Test Allowed Attachments";
+        String keyField = "Key";
+        String attachmentField = "Attachment Field";
+
+        log(String.format("Create a list '%s' with attachment field and auto-increment key.", listName));
+        _listHelper.createList(getProjectName(), listName, keyField,
+                new FieldDefinition(attachmentField, FieldDefinition.ColumnType.Attachment));
```

Comment:
Create list via API.
```suggestion
        new IntListDefinition(listName, keyField)
            .addField(new FieldDefinition(attachmentField, FieldDefinition.ColumnType.Attachment))
            .create(getProjectName(), createDefaultConnection());
```

Reply:
Done